src/plugins/data_source/README.md

```diff
@@ -1,9 +1,73 @@
-# data_source
+# DataSource Plugin
 
 An OpenSearch Dashboards plugin
 
-This plugin introduces OpenSearch data source into OpenSearch Dashboards, and provides related functions to connect to OpenSearch data sources.
+This plugin introduces multiple OpenSearch data sources into OpenSearch Dashboards, and provides related functions to connect to OpenSearch data sources.
 
+## Configuration
+Update the following flags in the opensearch_dashboards.yml file to apply changes.
+
+1. The dataSource plugin is by default disabled, to enable it:
+`data_source.enabled: true`
+
+2. The audit trail is by default enabled for logging the accessing to data source, to disable it:
+`data_source.audit.enabled: false`
+
+ - Current auditor configuration:
+```
+data_source.audit.appender.kind: 'file'
+data_source.audit.appender.layout.kind: 'pattern'
+data_source.audit.appender.path: '/tmp/opensearch-dashboards-data-source-audit.log'
+```
+
+3. The encryption related config are default to:
+```
+data_source.encryption.wrappingKeyName: 'changeme'
+data_source.encryption.wrappingKeyNamespace: 'changeme'
+data_source.encryption.wrappingKey: [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
+```
+Note that if any of the encryption keyring config value changed (wrappingKeyName/wrappingKeyNamespace/wrappingKey), all current encrypted credientails cannot be decrypted; Therefore, credentials of previously created data sources must be updated to continue use.
+
+How to generate random wrapping key for testing?  
```

Comment:
consider bold this line or make it a heading

Reply:
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2448/files#diff-cd490be76de9c19223cec880d293edc131e914c8ab82f5573232581a8d869d93R31-R32

Not very clear about the instuction. Suggest add more details such as:

What’s the best practice for generate secure wrapping key?
Ref: https://en.wikipedia.org/wiki/Cryptographically_secure_pseudorandom_number_generator

Reply:
**How to generate random wrapping key for testing?**  
Consider using an online generator or implementing customized logic. Read [more](https://en.wikipedia.org/wiki/Cryptographically_secure_pseudorandom_number_generator) about best practices for generating a secure wrapping key.

Thanks, how about this?

Reply:
> How to generate random wrapping key for testing?

I am not sure how much value with this question comparing with 'What’s the best practice for generate secure wrapping key?'. For me, security is more than just testing, and while we provide the default wrapping key, we strongly advise replacing it with more secure ones, such as cryptographic random numbers.

> online generator

I would not use that since it could easily acquired by attackers.

> implementing customized logic

This makes me feel like the feature is incomplete. 

I would prefer replace both querstion and anwser:

**What’s the best practice for generate secure wrapping key?**
Please read [more](https://en.wikipedia.org/wiki/Cryptographically_secure_pseudorandom_number_generator) about best practices for generating a secure wrapping key.

Reply:
Besides best practice, Do you have any recommendation on practical ways for users to generate the wrapping key?

Reply:
Yes there are a lot, such as Hardware security modules (HSM) or Key Management Service (KMS) from trust source.

Ref: 
1. https://csrc.nist.gov/publications/detail/sp/800-90a/rev-1/final
2. https://csrc.nist.gov/publications/detail/fips/140/2/finalh

Reply:
All the resources you shared are docs or algorithms. Does this mean the User has to implement customized logic to generate the wrapping key. Instead of a recommended tool?

Reply:
It's just random numbers, and they can figure out anything, even lotto numbers. It really depends on what level of secure they wanna achieve as the software user. However, we only provide **best practice** from our end.

Reply:
Synced offline, clarify the wrapping key for users, and leave the room for generation preference.